code/modules/vehicle/vehicle.dm

```diff
@@ -135,8 +138,9 @@
 		return 0
 	. = ..()
 	if(auto_door_open)
-		if(istype(M, /obj/machinery/door) && buckled_mob)
-			M.Bumped(buckled_mob)
+		if(istype(M, /obj/machinery/door) && buckled_mobs.len)
+			for(var/m in buckled_mobs)
+				M.Bumped(m)
```

Comment:
This can literally only cause problems if it ever happens, but I'm not sure how it'd be better